pallets/phala/src/mining.rs

```diff
@@ -288,18 +322,8 @@ pub mod pallet {
 			miner_info.cool_down_start = 0u64;
 			Miners::<T>::insert(&miner, &miner_info);
 
-			// Calcualte remaining stake
-			let v = FixedPoint::from_bits(miner_info.v);
-			let ve = FixedPoint::from_bits(miner_info.ve);
-			let return_rate = (v / ve).min(FixedPoint::from_num(1));
 			let orig_stake = Stakes::<T>::take(&miner).unwrap_or_default();
-			// If we consider kappa as a panelty of frequent exit:
-			// 	let tokenomic = Self::tokenomic();
-			// 	let returned = return_rate * orig_stake.to_fixed() * tokenomic.kappa();
-			let returned = return_rate * orig_stake.to_fixed();
-			// Convert to Balance
-			let returned = FixedPointConvert::from_fixed(&returned);
-			let slashed = orig_stake - returned;
+			let (_returned, slashed) = miner_info.calc_final_stake(orig_stake);
```

Comment:
since returned + slashed == orig_stake, we may only returned slashed in cal_final_stake